### PR TITLE
Adjustments to domain step in `onboarding-with-email` signup flow

### PR DESCRIPTION
**Pull request by fredrikekelund**
#### Proposed Changes

Letero is working on a set of improvements to the `onboarding-with-email` signup flow. This flow is not yet actively used, but it will capture users that have shown interest in the email solutions offered on WordPress.com. For more context, see pbLl1t-1f6-p2.

In this PR, I have made a few changes to the domains step in that signup flow:
1. Added a variation of the domains step: `domains-with-email`
2. Hidden the free wp.com subdomain suggestion in that step. We do this because we can't offer users email at all if they don't pick a custom domain first.
3. Changed the heading in that step to `Choose a domain for your Professional Email`

#### Testing Instructions

**1️⃣ Test the onboarding-with-email flow**
1. Navigate to `/start/onboarding-with-email`
2. Search for a domain
3. Ensure that you are not presented with a `Free` wordpress.com domain suggestion
4. Ensure that the heading reads `Choose a domain for your Professional Email`
5. Ensure that there is no `Choose my domain later` button in the sidebar
6. Click `Use a domain I own` in the sidebar
7. Ensure that the page matches the screenshot below (except that the step counter should say "Step 1 of 3"):

wordpress com_start_domains_use-your-domain(Medium laptop)

**2️⃣ Test the regular flow for regressions**
1. Navigate to `/start`
2. Search for a domain
3. Ensure that you *are* presented with a `Free` wordpress.com domain suggestion
4. Ensure that the heading reads `Choose a domain`
5. Ensure that there *is* a `Choose my domain later` button in the sidebar

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #

### CODE REVIEW

File: client/signup/steps/domains/index.jsx
Comment (wongasy) [inline client/signup/steps/domains/index.jsx]: I haven't done a thorough investigation, but it looks like we can define a new step to be utilized by the `onboarding-with-email` flow, rather than hardcoding the check here.

In `steps-pure.js`, the `domains` step is [configured](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/steps-pure.js#L342) as follows:
```
domains: {
			stepName: 'domains',
			apiRequestFunction: createSiteWithCart,
			providesDependencies: [
				'siteId',
				'siteSlug',
				'domainItem',
				'themeItem',
				'shouldHideFreePlan',
				'isManageSiteFlow',
			],
			optionalDependencies: [ 'shouldHideFreePlan', 'isManageSiteFlow' ],
			props: {
				isDomainOnly: false,
			},
			delayApiRequestUntilComplete: true,
		},
```

We can create a new step, say `mailbox-domain`,  based on `domains`, and configure `includeWordPressDotCom: false` as in `props`. (Note that the step name is reflected in the address bar, so at this step, the URL becomes `/start/onboarding-with-email/mailbox-domain`)

Then we can update the `onboarding-with-email` flow to replace the `domains` step with `mailbox-domain`:
https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/config/flows-pure.js#L158

Ideally, with a separate step defined, we can also customize the step title to say "Choose a domain for your Professional Email") by configuring the `stepSectionName` property. After a quick test, I see the the customize string does make its way to [getHeaderText()](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/index.jsx#L633). However, there appears to e an issue in the `DomainsStep` where it doesn't render when a custom title is supplied as a `prop`. @delputnam Is there any reason why we are preventing the `DomainsStep` component from rendering when a custom title is supplied?

**Review by wongasy — CHANGES_REQUESTED** [review]

File: client/signup/steps/domains/index.jsx
Comment (fredrikekelund) [reply client/signup/steps/domains/index.jsx]: Excellent feedback! 👍 Going with the approach you suggested.

I named the step `domains-with-email` in an attempt to align with how other `domain` variation steps are named. Sounds a bit awkward, but at least it's clear.

We can get back to the question about the header text in an upcoming PR.

File: client/signup/config/step-components.js
Comment (fredrikekelund) [inline client/signup/config/step-components.js:20]: I took the opportunity to sort these alphabetically

File: client/signup/steps/domains/index.jsx
Comment (fredrikekelund) [reply client/signup/steps/domains/index.jsx]: Actually, scratch that. It would make things easier to roll the update to the heading into this PR too.

Will update the PR once those changes are ready.

File: client/signup/config/step-components.js
Comment (wongasy) [reply client/signup/config/step-components.js:20]: I initially went with the exact same slug 😆 but it looked a bit strange in the address bar `start/onboarding-with-email/domains-with-email`, that's why I suggested `mailbox-domain`, but I suppose it isn't a huge deal :)

File: client/signup/config/steps-pure.js
Comment (wongasy) [inline client/signup/config/steps-pure.js:419]:
```suggestion
				forceHideFreeDomainExplainerAndStrikeoutUi: true,
```

File: client/signup/config/steps-pure.js
Comment (wongasy) [inline client/signup/config/steps-pure.js:431]:
```suggestion
				includeWordPressDotCom: false,
				isPlanSelectionAvailableLaterInFlow: false,
```

Comment (wongasy) [reply client/signup/config/steps-pure.js:431]: I checked data from the latest 2 months, the "Paid annual plan" sidebar widget doesn't generate any purchases of any product at all. Not only is the widget distracting users from their intention to setup Professional Email, it isn't doing any good in upsell. Let's remove it!

Screen Shot 2022-08-25 at 11 06 08 PM

**Comment by wongasy** [comment]
@fredrikekelund Changes are looking good! 

I found an issue where if user selects "Use a domain I own", they land on a screen that shows two headers that don't quite go well with each other:

Screen Shot 2022-08-25 at 11 16 59 PM

After user enters a domain, they land on the following screen:

Screen Shot 2022-08-25 at 11 23 12 PM

For simplicity, we can use the same approach in the domains flow where the header doesn't render in those screens, so that it just says "Use a domain I own". Provided that we already clearly specified the domain is used for setting up Professional Email, we should be good without the main header at the "Use a domain I own" step. 

Given the biggest hurdle of the flow is the requirement for users to select a paid plan, perhaps we should hold off rolling out the "Choose a Domain" changes until we complete the task for making plan selection optional. What do you think?

**Review by wongasy — CHANGES_REQUESTED** [review]

File: client/signup/config/steps-pure.js
Comment (fredrikekelund) [reply client/signup/config/steps-pure.js:431]: Good catch! But I actually opted for a different solution, which was to just hide the `Choose my domain later` button and keep the text.

My reasoning: the text is a pretty reasonable explainer, even more so since we will be changing the plans step to let users proceed with a free plan. If they do that, the domain will cost money once they reach checkout even if the domains step told them that the domain is `free for the first year`. By keeping this widget, we clarify the conditions for the domain being free. 

Then again, I guess you could argue that the plans page would _also_ probably clarify that the free plan will _not_ include a free domain for the first year. With this clarification, users shouldn't be completely dumbfounded when they reach checkout anyway. See this screenshot for clarification (the actual text would say the opposite, but I've circled the thing I'm talking about):

image-5

I'm open to hiding the entire widget as well, but I think that this approach does add value.

**Comment by fredrikekelund** [comment]
Thanks for the thorough review, @wongasy!

> I found an issue where if user selects "Use a domain I own"

I committed a fix for this that makes the `use-your-domain` section look identical to what we have on https://wordpress.com/start/domains/use-your-domain today. There are visual glitches on that page (bad padding on the heading, "unintentional" margin above the heading) that we could return to at a later time. Fixing those would benefit all signup flows.

> Given the biggest hurdle of the flow is the requirement for users to select a paid plan, perhaps we should hold off rolling out the "Choose a Domain" changes until we complete the task for making plan selection optional. What do you think?

Since this flow isn't being used yet, I think it's fine to finish up this PR and then work through the email and plan steps in upcoming PRs. In fact, I'm ready to submit a PR for the email step as soon as we've merged this one.

**Comment by wongasy** [comment]
@fredrikekelund 
>Since this flow isn't being used yet, I think it's fine to finish up this PR and then work through the email and plan steps in upcoming PRs. In fact, I'm ready to submit a PR for the email step as soon as we've merged this one.

The flow is actually being used, users can enter this flow via the existing Professional Email landing: https://wordpress.com/professional-email/

These improvements will unblock the launch of the new Emails landing page (we don't want to give a bad first impression by linking users to a flow that gives them a hard time when creating email), as well as improve the existing user journey from the Professional Email landing page :) The skippable PE step actually is less of the concern, but we should deploy this PR and the optional plan selection together.

File: client/signup/config/steps-pure.js
Comment (wongasy) [reply client/signup/config/steps-pure.js:431]: Yeah, I'm onboard with this approach!

File: client/signup/steps/domains/index.jsx
Comment (wongasy) [inline client/signup/steps/domains/index.jsx:660]: We have multiple instances of `use-your-domain` in this file, would it be a good opportunity to declare a constant to avoid use of multiple literal strings?

**Review by wongasy — CHANGES_REQUESTED** [review]
The PR looks good, except for the one minor feedback around the multiple instances of `use-your-domain`. 

And we should hold off merging this until we have the plan step updated: https://github.com/Automattic/wp-calypso/pull/66921#issuecomment-1228398193

In terms of the the PR for removing "skip" from the email step, if that change is dependent on this branch, perhaps we can do a PR against this branch to get review going?

File: client/signup/config/step-components.js
Comment (fredrikekelund) [reply client/signup/config/step-components.js:20]: Reevaluated this. It started to get a bit too weird when I was naming the `emails` step variation (`emails-with-email`..?)

So I went with your suggestion: the domains step is now `mailbox-domain` and the emails step will be `mailbox`

File: client/signup/steps/domains/index.jsx
Comment (fredrikekelund) [reply client/signup/steps/domains/index.jsx:660]: I'm a little bit hesitant to do that, just because there are similar assumptions in other files about the name of that section (e.g. [here](https://github.com/Automattic/wp-calypso/blob/4ccce1f/client/components/domains/use-your-domain-step/index.jsx#L96)). It would indeed be nice to define that slug in a single location, but unless we do a larger refactor to achieve a "single source of truth" across many files, I would be inclined to leave this as is.

**Comment by fredrikekelund** [comment]
> The flow is actually being used

My mistake! Although I think we're good to merge this first anyway.

> we should deploy this PR and the optional plan selection together.

Any special reason for that? 

I'm thinking that it would be convenient to merge this one first, then #67046, then the PR for the plans step. Why?
- Because I don't see any blockers for merging this PR (users that are coming from the [PE landing page](https://wordpress.com/professional-email/) should have basically the same expectations as those coming from the Email landing page)
- Because that's the order in which I worked on them
- Because there's a small dependency on this PR in #67046 (and the same thing will likely go for the plans selection PR)

**Review by AllTerrainDeveloper — APPROVED** [review]
Codewise looks good to me!
Test is now also working after latest patch.
BUT! I have a comment/suggestion, do we really need this button?

image

Since we already have one skip button in the bottom?

**Comment by fredrikekelund** [comment]
> do we really need this button?

It's set to be removed in #67046 👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7513638

Thank you @fredrikekelund for including a screenshot in the description! This is really helpful for our translators.

File: client/signup/config/steps-pure.js
Comment (wongasy) [inline client/signup/config/steps-pure.js]: @fredrikekelund We need to check for `en` locale before applying the fallback

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7513638

Thank you @fredrikekelund for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7513638).

**Pull request merged**
